storm-client/src/jvm/org/apache/storm/daemon/worker/WorkerState.java

```diff
@@ -377,15 +378,31 @@ public StormTimer getUserTimer() {
     public SmartThread makeTransferThread() {
         return workerTransfer.makeTransferThread();
     }
-    
+
+    public void suicideIfLocalAssignmentsChanged(Assignment assignment) {
+        if (assignment != null) {
+            Set<List<Long>> assignedExecutors = new HashSet<>(readWorkerExecutors(assignmentId, port, assignment));
+            if (!localExecutors.equals(assignedExecutors)) {
+                LOG.info("Found conflicting assignments. We shouldn't be alive!"
+                         + " Assigned: " + assignedExecutors + ", Current: "
+                         + localExecutors);
+                if (!ConfigUtils.isLocalMode(conf)) {
+                    suicideCallback.run();
+                } else {
+                    LOG.info("Local worker tried to commit suicide!");
+                }
+            }
+        }
+    }
+
     public void refreshConnections() {
         Assignment assignment = null;
         try {
             assignment = getLocalAssignment(stormClusterState, topologyId);
         } catch (Exception e) {
             LOG.warn("Failed to read assignment. This should only happen when topology is shutting down.", e);
         }
-
+        suicideIfLocalAssignmentsChanged(assignment);
```

Comment:
We can remove this from this PR since it is being addressed in your other PR https://github.com/apache/storm/pull/3291